utils/bazel/llvm-project-overlay/platform/BUILD.bazel

```diff
@@ -0,0 +1,64 @@
+# This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+"""Targetable platforms for the Bazel build."""
+
+# TODO(aaronmondal): Add triples for more useful platforms as per
+#                   `llvm/include/llvm/TargetParser/Triple.h`.
+
+platform(
+    name = "aarch64-unknown-linux-gnu",
+    constraint_values = [
+        "@platforms//cpu:aarch64",
+        "@platforms//os:linux",
+    ],
+)
+
+platform(
+    name = "arm64-apple-darwin",
+    constraint_values = [
+        "@platforms//cpu:arm64",
+        "@platforms//os:macos",
+    ],
+)
+
+platform(
+    name = "powerpc64le-unknown-linux-gnu",
+    constraint_values = [
+        "@platforms//cpu:ppc64le",
+        "@platforms//os:linux",
+    ],
+)
+
+platform(
+    name = "systemz-unknown-linux_gnu",
+    constraint_values = [
+        "@platforms//cpu:s390x",
+        "@platforms//os:linux",
+    ],
+)
+
+platform(
+    name = "x86_64-pc-win32",
+    constraint_values = [
+        "@platforms//cpu:x86_64",
+        "@platforms//os:windows",
+    ],
+)
+
+platform(
+    name = "x86_64-unknown-darwin",
```

Comment:
why is this not apple-darwin?

Reply:
This is taken 1:1 from https://github.com/llvm/llvm-project/blob/e5f5517f9121083dbb9d2841b607f504cbbb490b/utils/bazel/llvm-project-overlay/llvm/config.bzl#L101

I tried to minimize the difference in functionality between implementations. This triple seems wrong, but someone might depend on it and I thought it would be better to defer fixing this to another commit.

Reply:
After double-checking I think it was fine to change this after all. I changed it to `x86_64-apple-darwin`. During this I noticed that the windows target should be split into `x86_64-pc-windows-{gnu|msvc}` as well, but that's a bit trickier so I left a TODO to defer that for now.